### PR TITLE
Use HTS NFTs as if they're deployed ERC721/IERC721Metadata/IERC721Enumerable contracts

### LINKED ISSUE #79: HIP-218: Use HTS NFTs as if they're deployed ERC721/IERC721Metadata/IERC721Enumerable contracts
### Problem

As defined in: https://www.notion.so/limechain/HTS-precompiled-contract-functions-28619108abcd4d15a9f4539916e709ac we want to add additional support for the following functions:
- `symbol()`
- `tokenURI(uint256 _tokenId)`
- `ownerOf(uint256 _tokenId)`
- `approve(address _approved, uint256 _tokenId)`
- `setApprovalForAll(address _operator, bool _approved)`
- `isApprovedForAll(address _owner, address _operator)`

Usage: 
```
IERC721Metadata(token).name()
IERC721(token).getApproved(tokenId)
IERC721Enumerable(token).totalSupply()
```

### Solution

Add test support for the functions above.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #79 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/hts-precompile/erc-721/ERC721Contract.js [hunk 1]
```diff
@@ -44,6 +76,65 @@ describe("ERC721Contract tests", function () {
...
+    const isApprovedForAllBefore = await erc721Contract.isApprovedForAll(tokenAddress, erc721Contract.address, firstWallet.address);
+    const tx = await erc721Contract.setApprovalForAll(tokenAddress, firstWallet.address, true, {gasLimit: 1_000_000});
+    await tx.wait();
+    const isApprovedForAllAfter = await erc721Contract.isApprovedForAll(tokenAddress, erc721Contract.address, firstWallet.address);
+
+    expect(isApprovedForAllBefore).to.equal(false);
+    expect(isApprovedForAllAfter).to.equal(true);
+  });
+
+  it("should be able to execute transferFrom", async function () {
```
Comment (Nana-EC) [inline test/hts-precompile/erc-721/ERC721Contract.js:113]:
```suggestion
  it("should be able to execute delegate transferFrom", async function () {
```

File: test/hts-precompile/erc-721/ERC721Contract.js [hunk 2]
```diff
@@ -44,6 +76,65 @@ describe("ERC721Contract tests", function () {
...
+    const ownerBefore = await erc721Contract.ownerOf(tokenAddress, mintedTokenSerialNumber);
+    const erc721ContractNFTOwner = await ethers.getContractAt('ERC721Contract', erc721Contract.address, firstWallet);
+    await erc721ContractNFTOwner.delegateTransferFrom(tokenAddress, firstWallet.address, secondWallet.address, mintedTokenSerialNumber, {gasLimit: 1_000_000});
+    const ownerAfter = await erc721Contract.ownerOf(tokenAddress, mintedTokenSerialNumber);
+
+    expect(ownerBefore).to.equal(firstWallet.address);
+    expect(ownerAfter).to.equal(secondWallet.address);
+  });
+
+  it("should be able to approve", async function () {
```
Comment (Nana-EC) [inline test/hts-precompile/erc-721/ERC721Contract.js:127]:
```suggestion
  it("should be able to delegate approve", async function () {
```

**Review by Nana-EC — APPROVED** [review]
LG

**Review by georgi-l95 — APPROVED** [review]
LGTM!

**Pull request merged**